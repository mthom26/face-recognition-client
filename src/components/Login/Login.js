import React, { Component } from 'react';

const initialFormData = {
  email: '',
  password: '',
  confirmPassword: ''
};

class Login extends Component {
  constructor(props) {
    super(props);

    this.state = {
      formData: initialFormData,
      formSuccess: false,
      formFailure: false,
      loading: false
    };
  }

  onUpdateState = event => {
    const { target } = event;
    this.setState({
      formData: {
        ...this.state.formData,
        [target.name]: target.value
      }
    });
  };

  onSubmit = async event => {
    event.preventDefault();
    try {
      const url = `${process.env.REACT_APP_SERVER_URL}/login`;

      const result = await fetch(url, {
        method: 'POST',
        body: JSON.stringify(this.state.formData),
        headers: {
          'Content-Type': 'application/json'
        }
      });

      const data = await result.json();
      console.log(data);
    } catch (err) {
      console.log(err);
    }
  };

  render() {
    const { email, password, confirmPassword } = this.state.formData;

    return (
      <div>
        <h2>Login</h2>
        <form>
          <input
            onChange={this.onUpdateState}
            value={email}
            placeholder="Email"
            type="text"
            name="email"
            id="email"
          />

          <input
            onChange={this.onUpdateState}
            value={password}
            placeholder="Password"
            type="password"
            name="password"
            id="password"
          />

          <input
            onChange={this.onUpdateState}
            value={confirmPassword}
            placeholder="Confirm Password"
            type="password"
            name="confirmPassword"
            id="confirmPassword"
          />

          <button onClick={this.onSubmit}>Submit</button>
        </form>
      </div>
    );
  }
}

export default Login;
