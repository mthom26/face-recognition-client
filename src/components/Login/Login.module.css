.container {
  padding: 2rem;
  background: #00000076;
  display: flex;
  flex-direction: column;
  text-align: center;
}
