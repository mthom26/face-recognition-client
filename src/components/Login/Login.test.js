import React from 'react';
import { shallow } from 'enzyme';
import '../../setupEnzyme';

import Login from './Login';

describe('Login Component', () => {
  it('renders Login component', () => {
    expect(shallow(<Login />)).toMatchSnapshot();
  });
});
