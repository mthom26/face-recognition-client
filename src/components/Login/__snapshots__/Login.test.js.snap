// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component renders Login component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Login
        </h2>,
        <CustomForm
          onSubmitAction={[Function]}
          showEmail={true}
          showPassword={true}
          showPwConfirm={true}
        />,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login",
        },
        "ref": null,
        "rendered": "Login",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onSubmitAction": [Function],
          "showEmail": true,
          "showPassword": true,
          "showPwConfirm": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Login
          </h2>,
          <CustomForm
            onSubmitAction={[Function]}
            showEmail={true}
            showPassword={true}
            showPwConfirm={true}
          />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
          },
          "ref": null,
          "rendered": "Login",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onSubmitAction": [Function],
            "showEmail": true,
            "showPassword": true,
            "showPwConfirm": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
